docs/contributors/opentracingprovider.adoc

```diff
@@ -0,0 +1,203 @@
+
+* If references of type (a) exist, then if there is one reference it will be the primary. If there is more than one, then there is no distinct primary.
+
+* If there are no (a) references, then we check references of type (b) to see if they exist, and if so, if there is only one then it will be the primary. As previously, if there is more than one, then there is no distinct primary.
+
+* Finally if there are no (a) or (b) references, then we would check references of type (c). If one exists, it will be the primary, otherwise the is no primary.
+
+
+This approach means that references related to inbound communication (i.e. invocation of a service) take priority in terms of building a trace - which should be natural considering our principal aim is to build a distributed trace of communications between services.
+
+==== Processing a Primary Reference
+
+If a primary reference is found, then its processing will be dependent upon its type (i.e. a, b or c from previous section).
+
+===== Reference representing Extracted Trace State
+
+This reference represents potential extracted trace state propagated from an invoking client. Even if no trace state is actually found, it indicates that a communication has occurred to trigger this service.
+
+Therefore a new Trace fragment (and context) should be started, with the trace state (if available) initialised in the context and the top level Span should be represented by a Consumer node within that Trace fragment. The ‘id’ part of the trace state should be registered as a correlation id (of scope Interaction) on the Consumer.
+
+The other references should then be processed as discussed in the “Processing Remaining References” section below.
+
+===== Reference representing internal Parent/Child relationship of type ChildOf
+
+Create new node to represent the child span and add it as a child of the node associated with the supplied span.
+
+Share the trace context associated with the parent span.
+
+The other references should then be processed as discussed in the “Processing Remaining References” section below.
+
+===== Reference representing internal Parent/Child relationship of type FollowsFrom
+
+First step is to initialise a new Trace fragment (and context) and copy over information from the referenced trace context regarding the initial endpoint (URI & Operation) that were invoked for this service.
+
+The next step is to create a Consumer node as the top level node in this new fragment, with a CausedBy based correlation id using the nodeId associated with the referenced span. The URI and Operation fields on this Consumer node should be set to the same values associated with the initial endpoint copied over from the referenced trace context, and the endpointType of the Consumer should be set to null as it is an internal link. The timestamp on this Consumer should be set to the timestamp of the span.
```

Comment:
`within the initial endpoint` it can be confusing, maybe say the first span created from the extracted context